src-docs/src/views/modal/confirm_modal_loading.tsx

```diff
@@ -9,26 +9,25 @@ import {
 
 export default () => {
   const [isLoading, setIsLoading] = useState(false);
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [value, setValue] = useState('');
 
-  const searchTimeout = setTimeout(() => {
-    // Simulate a remotely-executed search.
-    setIsLoading(false);
-  }, 1200);
-
-  clearTimeout(searchTimeout);
+  let timeOutId: ReturnType<typeof setTimeout>;
+  const searchTimeout = () => setTimeout(() => setIsLoading(false), 1200);
```

Comment:
Wherever possible, we shouldn't lose comment context, especially since this one was useful in figuring out the intention of the code.
```suggestion
  // Simulate a remotely-executed search.
  const searchTimeout = () => setTimeout(() => setIsLoading(false), 1200);
```